config/initializers/paperclip.rb

```diff
@@ -0,0 +1,23 @@
+Paperclip.interpolates :attachments_root do |attachment, style|
+  if attachment.options[:storage] == :filesystem
+    "public/attachments"
+  else
+    "attachments"
+  end
+end
+
+Paperclip.interpolates :coverage_id do |attachment, style|
+  attachment.instance.coverage.id
```

Comment:
in this context, `attachment` means a wrapper around the actual file that is attached to our model. `instance` gets us from that back to the parent model instance (confusingly also called `Attachment` here...